### ISSUE TITLE
tips on evaluating stochastic policies / reproducibility

### ISSUE THREAD

**Issue by ronjamoller** [facebookresearch/habitat-api#409]
## ❓ Questions and Help
HI, I am trying to figure out whats the most reliable way to evaluate my trained (stochastic) policy. I am aware that with Neural Networks in general and stochastic behavior in particular this is always an issue but my task does not really fit a deterministic approach and going from the implementation in habitat_baselines which calls actor_critic.act with deterministic=False this evaluation is supported.
However, I would still like to maintain a certain level of reproducibility and looked into the relevant portion in the pytorch [documentation](https://pytorch.org/docs/stable/notes/randomness.html) - they suggested manually setting the torch seed, the numpy seed and the cudnn backend options. 
As a result, the success does not vary as much but it still varies, are there other settings to remove random influence ? I have noticed that even with the discussed settings the test episodes are not processed in the same order, there seems to be some sort of episode_id shuffle and i have not found the place in the code where that happens yet (only the scene shuffle), if someone could point me there i would appreciate it. There is also an episode seed, would this have to be manually set as well ?
tl;dr how to "de-randomize" an evaluation of a policy in habitat ?
thank you!

**Comment by dhruvbatra** [comment]
CC: @abhiskk -- could we document in a tutorial so it's addressed for future?
